Auto add plugins

jessestorbeck opened the pull request:
This PR includes two new features:

1. During `prep`, any plugins which are imported in experiment.js but haven't been added as dependencies get `yarn add`-ed.
2. During `install experiment`, users have the option of importing a jsPsych experiment.html if they choose the basic template. This will automatically identify the plugins they need and make a new experiment.js with their timeline and the required import statements. These plugins get added as dependencies later by the feature above.

I'm making this a draft, since (1) I'd like some help testing it and (2) we can wait to merge these changes until we finish with the other changes to `prep` re: jsPsych timeline assets.

jessestorbeck committed:
First try checking jsPsych packages during prep

jessestorbeck committed:
fix type error on string.match

jessestorbeck committed:
missing verbose argument

jessestorbeck committed:
added error handling

jessestorbeck committed:
add function to extract jsPsych timeline

jessestorbeck committed:
add function to extract necessary plugins

jessestorbeck committed:
add import names for jspsych plugins

jessestorbeck committed:
update handleInstall with jsPsych exp import

jessestorbeck committed:
update console.log

jessestorbeck committed:
spin off getJsPsychImports from  getJsPsychPlugins

jessestorbeck committed:
handle `yarn upgrade` to specific version

jessestorbeck committed:
simplify regex

jessestorbeck committed:
bug fix

jessestorbeck committed:
Adjust console logs and inquirer prompt

jessestorbeck committed:
bug fixes and log adjustments

jessestorbeck committed:
async issues

jessestorbeck committed:
jspsych import now entirely after getExpTemplate

jessestorbeck committed:
adjust experiment.js tabs

jessestorbeck committed:
remove tabs from expjs; figure out alignment later

jessestorbeck committed:
import experiment.html only for basic template 5+

hunterschep committed:
Update prep's index.js to copy necessaryassets to public

Copies everything from timeline folder if present into public/"nameofexp"

hunterschep committed:
updrate index.js to create experiments folder

jessestorbeck committed:
Merge branch 'prep-move-timeline' of https://github.com/pushkin-consortium/pushkin-cli into auto-add-plugins

jessestorbeck commented:
This now includes an additional feature. During `prep`, static assets used inside the jsPsych timeline in `experiment.js` are copied from the experiment's `web page/src/assets/timeline` folder to the site's `pushkin/front-end/public/experiments/[experiment_name]` folder.

`pushkin/front-end/public` can be referenced at runtime using the environment variable `process.env.PUBLIC_URL`.

hunterschep committed:
remove unneccessary overwrite

with short and longname already being overwritten, these lines are now not needed

jessestorbeck committed:
keep overwrite of longName

hunterschep committed:
update getExpTemplate to not log while waiting for a users input

Wrapping download and extraction process in a promise statement to stop premature logging

jessestorbeck committed:
reorg so getExpTemplate runs after all prompts

jessestorbeck committed:
await so logs don't come before prompts

jessestorbeck committed:
forgot to pass verbose to new functions

hunterschep reviewed (approved):
All new functionality and modules working when tested

jessestorbeck committed:
Adding verbose console logs

jessestorbeck committed:
another log

jessestorbeck committed:
adjust whitespace in experiment.js and console log

jessestorbeck committed:
additional error handling

jessestorbeck committed:
try again

jessestorbeck committed:
tweak error handling/logs

jessestorbeck committed:
tweak to getJsPsychTimeline

jessestorbeck committed:
bump version; fix readme image path

jessestorbeck committed:
typo/bug fixes

jessestorbeck committed:
hopefully last bug fix

jessestorbeck committed:
one more try at fixing exp yaml issue

jessestorbeck committed:
giving up on yaml name issue

jessestorbeck committed:
remove draft test
not sure how that got into this branch

hunterschep committed:
Fix prep

remove an erroneous return statement that was causing prepWeb to not be assigned a moduleName value

jessestorbeck committed:
deal with versions/tags in jspsych imports

jessestorbeck committed:
tweak version comment format

jessestorbeck committed:
typo fix - packages now upgrade reliably

jessestorbeck committed:
remove '@' from version number

jessestorbeck committed:
final tweak to version/tag format
